### ISSUE TITLE
Add `Reasoner::currenttask()`

### ISSUE THREAD

**Issue by ndarilek** [PsichiX/emergent#3]
I'm trying to integrate with Bevy. It doesn't seem like the existing task system will work as implemented, both because I can't access queries/commands in tasks but also because tasks don't behave like systems. Right now I have an `update` system that adds `current_state()` as a component, and each of my task systems checks the reasoner's state before running.

What I'd rather do instead is treat `Task` implementations as marker components. So instead of the reasoner running the task implementation, it'd simply expose its current task via a `current_task()` getter. I could then add the empty struct, then my task systems could use `With` constraints to only run when the specified component is added.

**Comment by PsichiX** [comment]
actually the problem you describe sounds like:
```rust
enum Behavior {
  Idle,
  Walk,
  Fight,
}

let reasoner = ReasonerBuilder::<_M, Behavior>::default()
  .state(Behavior, ReasonerState::new(_consideration, NoTask))
  .build();

if let Some(Behavior::Walk) = reasoner.active_state() {
  // do some work
}
```
what do you think about it?

using custom key type that can be read in the system, and NoTask that do no actual work, you make a decision maker that can do its work in the system depending on the key. key could be also used as component tag/marker.

**Comment by ndarilek** [comment]
Right, that's what I'm doing now. The problem is that `Behavior` is a single type. So you end up doing:

```
fn investigate(query: Query<(Entity, &Behavior, &Coordinates)>) {
    for (entity, behavior, coordinates) in query.iter() {
        if behavior == &Behavior::Investigate {
...
```

What I *want* to do is:

```
fn investigate(query: Query<(Entity, &Coordinates), With<Investigate>>) {
    for (entity, coordinates) in query.iter() {
```

As it stands, I don't see a way to use tasks in Bevy--at least not without having 99% of my game behavior in systems, and 1% in these odd traits called tasks. This makes it a bit more ergonomic to use systems for everything, and eliminates a layer of nesting/component access. Maybe there's a better pattern for this--it's just a bit odd having to perform this one conditional and layer of nesting when Bevy gives us `With` to make this easier to reason about.

Thanks.

**Comment by PsichiX** [comment]
now that you've explained that, that's a really good point. let me try to add and test that approach (i'm actually doing AI work for oxygengine and found related problems with tasks not being part of ECS itself, that i would like to address ASAP, and it seems this will solve our problems.
will be back with answers!
